docs/reference/ml/ml-shared.asciidoc

```diff
@@ -1113,6 +1113,27 @@ The field used to segment the analysis. When you use this property, you have
 completely independent baselines for each value of this field.
 end::partition-field-name[]
 
+tag::per-partition-categorization[]
+Settings related to how categorization interacts with partition fields.
+end::per-partition-categorization[]
+
+tag::per-partition-categorization-enabled[]
+If true, categorization is done separately for each value of the
+`partition_field_name` field. The default value is false. This setting can be
+set only when you create the job; it cannot be updated. This setting may only be
+true if every detector in the job that refers to `mlcategory` (the
+categorization output field) has the same value for `partition_field_name`.
+end::per-partition-categorization-enabled[]
+
+tag::per-partition-categorization-stop-on-warn[]
+This setting can be set to true only if per-partition categorization is enabled.
+If true, both categorization and subsequent anomaly detection stops for
+partitions where the categorization status changes to `warn`. This setting makes
```

Comment:
> where the categorization status changes to `warn`...

I don't think we talk anywhere else about when/why/where you'd see this status. Is this going to be added in a subsequent PR?

Reply:
> Is this going to be added in a subsequent PR?

Yes, for the per-partition aspect.

If you search for `tag::categorization-status` in https://github.com/elastic/elasticsearch/blob/master/docs/reference/ml/ml-shared.asciidoc you'll see it's currently a status that applies to the whole job.  Eventually, for jobs with per-partition categorization enabled there will be a separate categorization status per partition.  There will be subsequent PRs that implement this as the current ones are big enough already.  So I have slightly jumped the gun by mentioning it here.  The reason is because, as https://github.com/elastic/elasticsearch/pull/57683#pullrequestreview-424748586 alludes to, this PR already has to touch a huge number of files to make a very simple change.  If I had added the `enabled` and `stop_on_warn` settings in different PRs so that `stop_on_warn` actually worked then I would have needed two PRs that touched all these files.